### PR TITLE
Reenable survivor and xeno pregame changes

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Xenos will now take a upgrade penalty when dying pre-shutters and be teleported to a silo turf
image

Also fixes SSmonitor not setting gamestate
## Why It's Good For The Game

Survs are back, and maybe not shit this time

## Changelog
:cl:
add: Survs are back
add: Xenos dying before shutters open will have their upgrade points halved and be teleported to a silo
/:cl:

### CODE REVIEW

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #define COMSIG_GLOB_MOB_LOGIN "!mob_login"
 #define COMSIG_GLOB_MOB_LOGOUT "!mob_logout"
 #define COMSIG_GLOB_MOB_DEATH "!mob_death"
+	#define DEATH_STOPPED (1<<0)
```
Comment (XSlayer300) [inline code/__DEFINES/dcs/signals.dm:31]:
```suggestion
#define DEATH_STOPPED (1<<0)
```

File: code/datums/gamemodes/distress.dm [hunk 2]
```diff
@@ -286,3 +289,29 @@
...
+/datum/game_mode/infestation/distress/proc/on_preshutter_death(datum/source, mob/living/dier)
+	SIGNAL_HANDLER
+	if(!isxeno(dier))
+		return
+	if(SSmonitor.gamestate != SHUTTERS_CLOSED) //just in case
+		end_xeno_infspawns()
+		CRASH("Xeno attempted prespawn revive while Monitor gamestate dissalowed it")
+	var/mob/living/carbon/xenomorph/xeno = dier
+	xeno.revive()
+	xeno.visible_message("<span class='warning'> [xeno] contorts as the space around it tears and consumes it!</span>", "<span class='xenodanger'>YOU IGNORANT CHILD! I will keep you in this world if I must, but at your expense!</span>")
```
Comment (XSlayer300) [inline code/datums/gamemodes/distress.dm:303]:
```suggestion
	xeno.visible_message("<span class='warning'>[xeno] contorts as the space around it tears and consumes it!</span>", "<span class='xenodanger'>YOU IGNORANT CHILD! I will keep you in this world if I must, but at your expense!</span>")
```

File: code/__DEFINES/dcs/signals.dm (hunk 1 above)
Comment (BraveMole) [reply code/__DEFINES/dcs/signals.dm:31]: Nop, that's standard for bit flag sent by signals

**Comment by Jeff-jefferson-jeffery** [comment]
This makes survivor completely pointless to play. Why reenable it if they are unable to make an impact on the round?
Survivor is my favorite role and this is completely wack, on any given round as one, you cannot hold a position because you do not have enough ammo nor enough storage space to carry it, and this will only make that problem worse as no matter how many you kill, you will never impact their numbers.
Of course, you also can't hunt xenos like this because you can't well... kill anything.

A better soloution would be reducing the armour on their suits to light jeager levels, instead of being heavy jeager with no slowdown like it is currently, and disabling leaderless collapse before shutters open, this fixes the "issue" with survivor majors, as no matter what they won't be able to loose to a survivor, but will be able to be impacted by them.

File: code/__DEFINES/dcs/signals.dm (hunk 1 above)
Comment (TiviPlus) [reply code/__DEFINES/dcs/signals.dm:31]: ^^

**Comment by TiviPlus** [comment]
Because if they have an impact on the round they'll only be played by people powergaming to the max to try kill xenos as fast as they can

**Comment by Jeff-jefferson-jeffery** [comment]
You overestimate the ability of survivors - if the xenos are even mildly competent, they can deal with survivors, it worked like that before and it will work like that now, even moreso considering shit like xeno turrets.
If you don't like people "powergaming to the max", just remove pulse rifles and AKs from armories, as those are their best weapons, by far.

**Comment by TiviPlus** [comment]
Seeing what happened I no doubt that what happened once will happen again and "Just nerf it lmao" Will make it into a good-player-exclusive role only

**Comment by Jeff-jefferson-jeffery** [comment]
I am the guy behind the quad survivor major streak. It was only possible because of a combination of busted game mechanics (pools barely giving xenos any larva, pool collapse before shutters while xenos can't get any points), Completely inept xeno leaders and the strong gear available to you on certain maps.
In the current state of the game - it would require a far better player than me, or all hive leaders to go AFK outside of caves in order to pull off anything similar, due to the existence of xeno turrets that survivors do not have any of the "good" counters to.
A "good player only role" is a shitty argument - Survivor, inherently requires you to be a "good player", you need to be able to fend off t1-t2 xenos alone or in a small group, and you need to know the map well, not much can change that because it's just the nature of the role. Newer players could still use it to learn groundside maps and get use to fighting xenos alone.

**Comment by Tennessee116** [comment]
How about instead of making every survivor Rambo, we slightly revert and give them some job-based equipment. Specifically, give them some cool gimmicky thing marines can sell through requisitions for lots of points.

Or maybe bring them back when there is some sort of substantial R&D system where they know some wacky xenomorph science that researchers can use to progress the tree.

~~Or just become a TDM server under the guise of MRP.~~

**Comment by SilencedMP5A5** [comment]
if you get killed by a survivor you deserve the death lol

**Comment by SkeletalElite** [comment]
What happens if there is no silos

**Comment by Jeff-jefferson-jeffery** [comment]
I have seen plenty of newer players playing survivor - sure, most of them just hide in a locker and die, but i've had a couple of them fight and hold with me and come out "successful", it's a rare occurrence, but it happens.
If survivors do exist, they should have the power to impact the round because why play them otherwise? With these changes, absolutely nothing you do as survivor matters, even if you wipe the hive, they barely get a slap on the wrist.

Survivors will not be able to "win" if you remove the leaderless collapse Pre shutters, and frankly, if all xeno leaders die to survivors, they really deserved it.

Marines having prep time wasted is not an issue with survivor - it's an issue with the game as a whole, rounds are being made shorter and shorter (most marine majors happen in under 20m now) while no one has touched prep time. 

I just cannot see these changes bringing survivors back in any fun way - it will just result in xenos bumrushing them because the survivor literally cannot kill them.

**Comment by AyyRobotics** [comment]
Yeah this doesn't seem like the best way to re-add survs. Xenos are just gonna be more brainless since they can't die while fighting them, and survs are probably gonna be played less often since they know that they can do nothing to the xenos. Unless the xenos are weaker to compensate then I can't see many people playing survivor. Survs in themselves are difficult roles but when you combine an already difficult role with unkillable enemies, the challenge of it is so astronomical that not even the survivor gods would want to play it. All this is gonna seem to do is give xenos a couple of free bodies to Psydrain for little cost.

If we really want to make survivors something different than just a role to kill xenos early, then why not put more aspect into the "surviving" part of it. Let them hide under tables like runners, be immune to xenovision so they actually can hide and run away, give them weapons like the powerfist that are more "Fuck-off" tools than weapons. This would make them actually interesting and unique, while still being able to do something to xenos if they have to. Tennessee's idea is also a pretty good one.

And fuck it, if a couple of survs want to rush the armory and grab some shotguns to go unga the entire hive, then let them. Xenos have all the tools they need to fight off much bigger forces than two colonists with shotguns, so if they die in a way that significantly impacts the round, that's on them.

**Comment by SplinterGP** [comment]
i'd say this is also kek because it allows xenos to get more psych points with surv bodies with new beno stuff so weh, i don't think its a good idea to do it this way.

**Comment by bowlofcereal** [comment]
Why even bother adding them back if anything they do does not matter in any way? Xeno attacks him and he kills it, and then it just teleports right back to a silo to try it again. The fucks the point in playing it if xenos have 0 ramifications for just running after you? Halving upgrade points ain't shit. Don't even bother adding survivors back with this kind of change.

TL;DR if you die to a survivor youre a fucking idiot and thats on you

**Comment by Swagile** [comment]
if you do things like this you'd have to give survivors a small outpost on every map thats a one way entrance with 2 base turrets pointed at it, a very small medbay, small engi bay, and small prep area with 1 surv leader with all skills, 1 doctor, 1 engi, and 2 standard dudes with shutters that auto open in 5 mins on the outpost so that survs stand a chance to have some fun and combat xenos can have fun attacking it with minor penalty

but that takes a lot of work for little gain

idfk, i just don't want survivor coming back in this state, cause its shit

**Comment by EgorDinamit** [comment]
I'd honestly just lower their armour values/give them a slowdown, so they can't be fast death-machines.

**Comment by spookydonut** [comment]
> 
> 
> my go would rather make stealth gear, which i think its hard to make due to byond limitations on view stuff, or make them hold a position, or make them dying make them respawn with a droppod from sky or something like that. otherwise meh, doenst make sense to bring them back like this so.

agreed

**Comment by Tennessee116** [comment]
A second idea, turn them into freelancers that just have RNG equipment and can't use marine equipment.
